src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -1101,9 +1101,15 @@ def get_probe(self):
     def get_channel_locations(self) -> np.ndarray:
         # important note : contrary to recording
         # this give all channel locations, so no kwargs like channel_ids and axes
-        all_probes = self.get_probegroup().probes
-        all_positions = np.vstack([probe.contact_positions for probe in all_probes])
-        return all_positions
+        probegroup = self.get_probegroup()
+        probe_as_numpy_array = probegroup.to_numpy(complete=True)
+        # we need to sort by device_channel_indices to ensure the order of locations is correct
+        probe_as_numpy_array = probe_as_numpy_array[np.argsort(probe_as_numpy_array["device_channel_indices"])]
```

Comment:
This correct but we have a internal problem that should be solved about this.
Becasue the contact_vector should be always ordered I think.
In recording this is the case but no in analyzer because we have ProbeGroup instead of the contact_vector representation